tmt/steps/execute/scripts/tmt-report-result

```diff
@@ -11,6 +11,61 @@ help=False
 die() { echo "$*" >&2; exit 2; }  # complain to STDERR and exit with error
 needs_arg() { if [ -z "$OPTARG" ]; then die "No arg for --$OPT option"; fi; }
 
+write_report_file () {
+    rm -f $REPORT_RESULT_OUTPUTFILE
+    echo "SERVER=$server" >> $REPORT_RESULT_OUTPUTFILE
+    echo "PORT=$port" >> $REPORT_RESULT_OUTPUTFILE
+    echo "MESSAGE=$message" >> $REPORT_RESULT_OUTPUTFILE
+    echo "OUTPUTFILE=$outputFile" >> $REPORT_RESULT_OUTPUTFILE
+    echo "DISABLEPLUGIN=$disablePlugin" >> $REPORT_RESULT_OUTPUTFILE
+    echo "TESTNAME=$TESTNAME" >> $REPORT_RESULT_OUTPUTFILE
+    echo "TESTRESULT=$TESTRESULT" >> $REPORT_RESULT_OUTPUTFILE
+    echo "METRIC=$METRIC" >> $REPORT_RESULT_OUTPUTFILE
+}
+
+check_report_file () {
+    declare -A RESULTS_HIERARCHY=( ["FAIL"]=4 ["WARN"]=3 ["PASS"]=2 ["SKIP"]=1 )
+    # Create the output file in which to store the results.
+    REPORT_RESULT_OUTPUTFILE="$TMT_TEST_DATA/restraint-result"
+    if [ -e "$REPORT_RESULT_OUTPUTFILE" ]; then
+        # Check the existing output file.
+        # If new result is higher on the hierarchy
+        # or no result is contained in the file
+        # then overwrite the file with the new
+        # report contents.
+        fileResult=$(/usr/bin/grep "TESTRESULT" $REPORT_RESULT_OUTPUTFILE |
+        cut -d = -f 2)
+        if [ ! $fileResult ]; then
+            write_report_file;
+            exit 0
+        elif [ ! $TESTRESULT ]; then
+            exit 0
+        fi
+        fileHierarchy="${RESULTS_HIERARCHY[$fileResult]}"
+        # Compare the current result hierarchy against
+        # the hierarchy result retrieved from the file.
+        # Overwrite the file if the current hierachy is higher.
+        thisResultHierarchy="${RESULTS_HIERARCHY[$TESTRESULT]}"
+        if [ $thisResultHierarchy -gt $fileHierarchy ]; then
+            write_report_file
+            exit 0
+        fi
+        exit 0
+    else
+        # Create the output file with the report contents.
+        write_report_file
+        exit 0
+    fi
+}
+
+if [[ "$0" == *"rhts-report-result"* ]]; then
+    TESTNAME=$1
+    TESTRESULT=$2
+    outputFile=$3
+    METRIC=$4
+    check_report_file
```

Comment:
Don't we call check_report_file twice? The final one (line 137) is executed as well.

Reply:
The function 'check_report_file' results with the script exiting.

This BASH script deals with two paths: rhts & restraint. rhts requires less processing, and so the 'check_report_file' function can be called earlier to exit execution, whereas restraint requires the script to continue on through the additional logic of the 'check_opt_args' function before finally executing the 'check_report_file' function to exit. If the rhts path does not call the 'check_report_file' function then the script continues to execute and the variable assignment for the contents in the 'restraint-result' file end up incorrect.